SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -21,62 +21,113 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
 
 void SpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker_ComponentUpdateOp& Op)
 {
-	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
-	if (StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
+	if (CanEnforce(Op.entity_id))
 	{
-		QueueAclAssignmentRequest(Op.entity_id);
+		MaybeQueueAclAssignmentRequest(Op.entity_id);
 	}
 }
 
-// This is called whenever this worker becomes authoritative for the ACL component on an entity.
-// It is now this worker's responsibility to make sure that the ACL reflects the current intent,
-// which may have been received before this call.
-void SpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+void SpatialLoadBalanceEnforcer::OnLoadBalancingComponentAdded(const Worker_AddComponentOp& Op)
 {
-	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
-		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	if (!(Op.data.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || Op.data.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID))
```

Comment:
```suggestion
	if (Op.data.component_id != SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID && !Op.data.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID))
```

Reply:
Doesn't `!Op.data.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID` need to be `Op.data.component_id != SpatialConstants::ENTITY_ACL_COMPONENT_ID`

Reply:
That's what it says (🤫🤫🤫)

Reply:
Also, prefer to wrap `op.data.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID` in a check.

We should also just `return;` for non-authoritative op, losing authority over EntityACL should be fine?

Reply:
made these all checks